### PR TITLE
feat: when publishing a managed gateway, shard gateway will no longer…

### PR DESCRIPTION
**Pull request by zhu327**
… be published at the same time.

### Description

发布专享网关时不再同时发布共享网关

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [x] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: src/dashboard/apigateway/apigateway/apps/release/releasers.py [hunk 1]
```diff
@@ -293,7 +293,14 @@ def _create_release_task_for_micro_gateway(self, release: Release, release_histo
         )  # type: ignore
 
     def _create_release_tasks(self, release: Release, release_history: ReleaseHistory):
-        for fn in [self._create_release_task_for_shared_gateway, self._create_release_task_for_micro_gateway]:
+        # NOTE: 发布微网关时不再同时发布专享网关
+        micro_gateway = release.stage.micro_gateway
+        if not micro_gateway or micro_gateway.is_shared:
+            functions = [self._create_release_task_for_shared_gateway]
+        else:
+            functions = [self._create_release_task_for_micro_gateway]
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/apps/release/releasers.py:301]: 只剩一个的话, 不要搞 functions=[], 直接 fn=

File: src/dashboard/apigateway/apigateway/controller/distributor/combine.py [hunk 2]
```diff
@@ -46,13 +46,18 @@ def foreach_distributor(
         assert micro_gateway.is_shared
 
         managed_micro_gateway = stage.micro_gateway
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/controller/distributor/combine.py:48]: 补充注释, 降低这里的理解成本(未来得重构....)
- micro_gateway.is_shared 代表什么
- not managed_micro_gateway 意味着什么
- include_gateway_global_config 会多做什么, 为什么line 50是False, line63是 True
- generate_chart  为什么这里是False

Comment (zhu327) [reply src/dashboard/apigateway/apigateway/controller/distributor/combine.py:48]: include_gateway_global_config 还没完全看懂, 先不加

**Review by alex-smile — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/apps/release/releasers.py [hunk 3]
```diff
@@ -293,7 +293,14 @@ def _create_release_task_for_micro_gateway(self, release: Release, release_histo
         )  # type: ignore
 
     def _create_release_tasks(self, release: Release, release_history: ReleaseHistory):
-        for fn in [self._create_release_task_for_shared_gateway, self._create_release_task_for_micro_gateway]:
+        # NOTE: 发布微网关时不再同时发布专享网关
```
[the remark this answers is not in the document]
Comment (wklken) [reply src/dashboard/apigateway/apigateway/apps/release/releasers.py:296]: 更新一下

**Comment by Han-Ya-Jun** [comment]
gpg 好像失效了

**Review by Han-Ya-Jun — APPROVED** [review]

**Pull request merged**